File: MATTHEW/leetcode/AIcoding/2.Gemini-2.0/java_solutions/3131-find-the-integer-added-to-array-i.java
class Solution {
    public int getSumAbsoluteDifferences(int[] nums) {
        int n = nums.length;
        int[] result = new int[n];
        int sum = 0;
        for (int num : nums) {
            sum += num;
        }
        int leftSum = 0;
        for (int i = 0; i < n; i++) {
            int rightSum = sum - leftSum - nums[i];
            result[i] = nums[i] * i - leftSum + rightSum - nums[i] * (n - 1 - i);
            leftSum += nums[i];
        }
        return result;
    }
}

File: leetcode/AIcoding/2.Gemini-2.0/rust_solutions/3131-find-the-integer-added-to-array-i.rs
impl Solution {
    pub fn add_to_array_form(num: Vec<i32>, k: i32) -> Vec<i32> {
        let mut result: Vec<i32> = Vec::new();
        let mut carry = 0;
        let mut k_mutable = k;
        let mut i = num.len() as i32 - 1;

        while i >= 0 || k_mutable > 0 || carry > 0 {
            let mut sum = carry;
            if i >= 0 {
                sum += num[i as usize];
            }
            if k_mutable > 0 {
                sum += k_mutable % 10;
                k_mutable /= 10;
            }

            result.push(sum % 10);
            carry = sum / 10;

            if i >= 0 {
                i -= 1;
            }
        }

        result.reverse();
        result
    }
}